src/playground/services/session/register/endpoint.clj

```diff
@@ -0,0 +1,30 @@
+(ns playground.services.session.register.endpoint
+  (:require [clojure.java.jdbc :as jdbc]
+            [clojure.spec.alpha :as spec]
+            [honeysql.core :as h]
+            [ring.util.response :as ring-resp]
+            [io.pedestal.http.route :refer [url-for]]
+            [buddy.hashers :as hashers]
+            [playground.services.session.register.logic :as logic]))
+
+
+(spec/def ::username (spec/and string? seq))
```

Comment:
Don't duplicate equivalent specs. Create something like `src/playground/models/user.clj`: a ns dedicated to the User model  (model -> MVC terminology)

You define the user specs there, and the other 2 ns's consume them.